device_backends/LogicalNameMapping/src/LogicalNameMappingBackend.cc

```diff
@@ -215,7 +218,7 @@ namespace ChimeraTK {
         target_info = _catalogue_mutable.getRegister(lnmInfo.registerName);
         // target_info might also be affected by plugins. e.g. forceReadOnly plugin
         // we need to process plugin list of target register before taking over anything
-        LNMBackendRegisterInfo& i = static_cast<LNMBackendRegisterInfo&>(target_info.getImpl());
+        auto& i = dynamic_cast<LNMBackendRegisterInfo&>(target_info.getImpl());
         for(auto& plugin : i.plugins) {
```

Comment:
The logic makes sure that the cast always succeeds. If it would fail, it does not throw bad_cast. It returns nullptr if the dynamic cast does not succeed. So to be complete, one would have to check for that.

The linter complained. I basically was lazy because letting the IDE replace the cast was one click, and silencing the linter would have involved manual typing. This is not a performance critical section and I wanted as little linter exceptions as possible. (I also commented each one why it is done)

Reply:
It cannot return a nullptr, this is a reference. bad_cast is the only way to signify that this cast failed. see https://en.cppreference.com/w/cpp/language/dynamic_cast, 5c)

Reply:
Ah, right. It's not a pointer cast. What do we do? Leave it, or revert and silence the linter? Or catch and convert to ChimeraTL::logic_error?

Reply:
as discussed: Leave the bad_cast escaping, points to programming error.